docs/proposals/security-context-constraints.md

```diff
@@ -0,0 +1,320 @@
+
+As reusable objects in the root scope, security context constraints follow the lifecycle of the
+cluster itself.  Maintenance of constraints such as adding, assigning, or changing them is the
+responsibility of the cluster administrator.  For
+instance, creating a new user within a namespace should not require the cluster administrator to
+define the user's security context constraints.  They should receive the default set of constraints
+that the administrator has defined for users.
+
+Finalizers should ensure that any user or group references in a security context constraint object
+will be removed when the referenced object is removed.
+
+
+## Default Security Context Constraints And Overrides
+
+In order to establish security context constraints for service accounts and users there must be a way
+to identify the default set of constraints that is to be used.  This is best accomplished by using
+groups.  As mentioned above, groups may be used by the authentication/authorization layer to ensure
+that every user maps to at least one group (with a default example of `system:authenticated`) and it
+is up to the cluster administrator to ensure that a security context constraint object exists that
+references the group.
+
+If an administrator would like to provide a user with a changed set of security context permissions
+they may do the following:
+
+1.  Create a new security context constraints object and add a reference to the user or a group
+that the user belongs to.
+1.  Add the user (or group) to an existing security context constraints object with the proper
+elevated privileges.
+
+## Admission
+
+Admission control using an authorizer allows the ability to control the creation of resources
+based on capabilities granted to a user.  In terms of the security context constraints it means
+that an admission controller may inspect the user info made available in the context to retrieve
+and appropriate set of security context constraints for validation.
+
+The appropriate set of security context constraints is defined as all of the security context
+constraints available that have reference to the user or groups that the user belongs to.
+
+Admission will use the SecurityContextConstraintsAllocator to ensure that any requests for a
+specific security context constraint setting or to generate settings using the following approach:
+
+**Pod with referenced service account**
+
+1.  First, ensure that the user has the ability to use the security context constraints of the
+service account.  This can be accomplished by finding the intersection of allowed constraints between
+the user and the service account.  If none exists then the user does not have permissions to
+run as the service account.
+2.  Ask the SecurityContextConstraintsAllocator to create the security context.  The allocator
+will not overwrite fields already set in the container.
+3.  Validate that the generated SecurityContext falls within the boundaries of the security context
```

Comment:
I don't see what the point is of double-checking what the admission controller does.  It is trusted to generate values for this field, right, so why double check?